query/sources/impl/src/main/java/org/codice/ddf/admin/sources/commons/utils/SourceValidationUtils.java

```diff
@@ -20,48 +20,52 @@
 
 import org.codice.ddf.admin.common.fields.base.scalar.StringField;
 import org.codice.ddf.admin.common.report.ReportImpl;
-import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.internal.admin.configurator.actions.ServiceReader;
 
 import ddf.catalog.service.ConfiguredService;
 import ddf.catalog.source.Source;
+import ddf.catalog.util.Describable;
 
 public class SourceValidationUtils {
 
     /**
      * Determines whether the service configuration identified by the service pid has the same name as sourceName.
      * If the service configuration's id equals the sourceName return {@code true}, otherwise {@code false}.
      *
-     * @param servicePid service pid of the service configuration
-     * @param sourceName source name to check against existing source name
-     * @param configuratorFactory configurator factory to get services
+     * @param servicePid    service pid of the service configuration
+     * @param sourceName    source name to check against existing source name
+     * @param serviceReader service to read service state
      * @return {@code true} if the sourceName matches the existing configuration's id, {@code false} otherwise
      */
-    public static boolean hasSourceName(String servicePid, String sourceName, ConfiguratorFactory configuratorFactory) {
-        Source source = getAllSourceReferences(configuratorFactory).stream()
+    public static boolean hasSourceName(String servicePid, String sourceName,
```

Comment:
Can you leave a todo here to remove these static methods and pass the `ServiceReader` through the constructor please